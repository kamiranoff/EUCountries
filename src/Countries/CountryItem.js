import React from 'react';
import PropTypes from 'prop-types';
import { Text, View, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  card: {
    backgroundColor: '#eee',
    margin: 10,
    padding: 10,
  },
});

const CountryItem = ({ item }) => (
  <View style={styles.card}>
    <Text style={{ fontWeight: '600' }}>{item.name}</Text>
    <Text>{item.subregion}</Text>
    <Text>{item.capital}</Text>
    <Text>Lang: {item.languages[0].name}</Text>
  </View>
);

CountryItem.defaultProps = {
  item: {
    name: '',
    subregion: '',
    capital: '',
    languages: [{ name: '' }],
  },
};

CountryItem.propTypes = {
  item: PropTypes.shape({
    name: PropTypes.string,
    subregion: PropTypes.string,
    capital: PropTypes.string,
    languages: PropTypes.array,
  }),
};

export default CountryItem;
