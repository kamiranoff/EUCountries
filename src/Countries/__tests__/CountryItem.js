import React from 'react';
import renderer from 'react-test-renderer';
import CountryItem from '../CountryItem';

describe('CountryItem', () => {
  it('should render the component', () => {
    const wrapper = renderer.create(<CountryItem />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
