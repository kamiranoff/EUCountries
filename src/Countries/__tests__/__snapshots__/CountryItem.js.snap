// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryItem should render the component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eee",
      "margin": 10,
      "padding": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "600",
      }
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Lang: 
    
  </Text>
</View>
`;
